src/aia_clic_extension.adoc

```diff
@@ -0,0 +1,1816 @@
+
+.A pending-and-enabled interrupt _i_ causes the hart to resume execution if interrupt _i_
+* has the same privilege mode as the current privilege mode and
+* the interrupt priority reduction tree selects interrupt _i_ as the maximum across all pending-and-enabled interrupts and
+* the interrupt _i_ level is greater than max({intstatus}.{il}, {intthresh}.`th` )
+
+.A pending-and-enabled interrupt _i_ causes the hart to resume execution if interrupt _i_
+* has a lower privilege mode than the current privilege mode and
+* the interrupt priority reduction tree selects interrupt _i_ as the maximum across all pending-and-enabled interrupts and
+
+NOTE: {intthresh} only applies to the current privilege mode.  There is a proposal to add a new WFMI instruction ("wait for mode's interrupts") to the privilege specification. This instruction only has to wakeup for pending-and-enabled interrupts in the current mode, and is not required to wakeup for pending-and-enabled interrupts in lower privilege modes. Pending-enabled higher privilege-mode interrupts will interrupt/wakeup as usual.
+
+==== Synchronous Exception Handling
+
+Horizontal synchronous exception traps, which stay within a privilege
+mode, are serviced with the same interrupt level as the instruction
+that raised the exception.
+
+Vertical synchronous exception traps, which are serviced at a higher
+privilege mode, are taken at interrupt level 0 in the higher privilege
+mode.
+
+WARNING: Traps should be avoided at any time when {epc}/{cause} are live
+because these CSRs will be overwritten. Software should try to back them
+up if needed.
+
+==== Non-Resumable Non-Maskable Interrupts
+
+The handling of NMIs is implementation-specific, but NMIs are always
+handled in M-mode and can overwrite `mepc` and `mcause` of an active
+M-mode CLIC interrupt handler.
+
+==== Resumable Non-Maskable Interrupts
+
+NOTE: This section describes the interaction of the CLIC with the
+proposed new RNMI specification.
+
+The resumable NMI (RNMI) extension adds additional `mnepc`, `mncause`,
+and `mnstatus` CSRs.  The RNMI CSRs are separate from the CLIC CSRs and so CLIC state does not need to be saved in RNMI CSRs.
+This makes the design of the RNMI extension orthogonal to the hart's general interrupt scheme.
+
+==== Returns from Handlers
+
+As stated in the RISC-V Privilege Specification,
+when executing an xRET instruction, supposing xPP holds the value y, xIE is set to xPIE; the
+privilege mode is changed to y; xPIE is set to 1; and xPP is set to the least-privileged supported
+mode.  The {ret} instruction sets the pc to the value stored in the {epc} register.
+Additionally in CLIC mode, {ret} sets {intstatus}.{il} to {pintstatus}.{pil}.
+The {ret} instruction does not modify the {pintstatus}.{pil} field in {pintstatus}.
+
+
+=== CLIC Level Control - Interrupts at machine level
+
+[source]
+----
+       Number  Name         Description
+ (NEW) 0x346   mpintstatus  Previous interrupt levels

```

Comment:
this is more than levels. should be interrupted context or something (if we remove exccode)